File: solutions/28. Find the Index of the First Occurrence in a String/28.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        gap = len(needle)
        if needle not in haystack:
            return -1 
        else :
            for i in range(len(haystack)):
                if haystack[i:i+gap] == needle: return i 

File: solutions/28. Find the Index of the First Occurrence in a String/28.rb
# @param {String} haystack
# @param {String} needle
# @return {Integer}
def str_str(haystack, needle)
    gap = needle.size
    return -1 unless haystack.include? needle
    haystack.size.times do |i|
        return i if haystack[i, gap] == needle
    end
end